src/reduce.test.js

```diff
@@ -11,9 +11,41 @@ describe('reduce', () => {
     delete Array.prototype.reduce2;
   });
 
-  it('should ', () => {
+  it(`should return error if callback doesn't passed`, () => {
+    expect(() => {
+      [1, 2, 3].reduce2();
+    }).toThrow();
+  });
+
+  it('should work with undefined passed as an initial value', () => {
+    const callback = (a, b) => a + b;
+
+    expect([1, 2, 3].reduce2(callback, undefined)).toBe(NaN);
```

Comment:
The expectation that `[1, 2, 3].reduce2(callback, undefined)` should return `NaN` is incorrect. When `undefined` is passed as the initial value, the first element of the array should be used as the initial accumulator value, and the result should be the sum of the array elements, which is `6`.